### PR TITLE
Update REST API security related BBEs

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#3171](https://github.com/ballerina-platform/ballerina-standard-library/issues/3171)

### CODE REVIEW

File: examples/http-service-basic-auth-file-user-store/http_service_basic_auth_file_user_store.bal [hunk 1]
```diff
@@ -12,9 +12,8 @@ listener http:Listener securedEP = new(9090,
 // The service can be secured with Basic Auth and can be authorized optionally.
 // Using Basic Auth with the file user store can be enabled by setting the
 // [`http:FileUserStoreConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/FileUserStoreConfig) configurations.
-// Authorization is based on scopes. A scope maps to one or more groups.
-// Authorization can be enabled by setting the `string|string[]` type
-// configurations for `scopes` field.
+// Authorization is based on scopes. A scope maps to one or more groups. Authorization can be 
+// enabled by setting the `string|string[]` type configurations for `scopes` field.
```
Comment (praneesha) [inline examples/http-service-basic-auth-file-user-store/http_service_basic_auth_file_user_store.bal:16]:
```suggestion
// enabled by setting the `string|string[]` type configurations for the `scopes` field.
```

File: examples/http-service-basic-auth-ldap-user-store/http_service_basic_auth_ldap_user_store.bal [hunk 2]
```diff
@@ -12,9 +12,8 @@ listener http:Listener securedEP = new(9090,
 // The service can be secured with Basic Auth and can be authorized  optionally.
 // Basic Auth using the LDAP user store can be enabled by setting the
 // [`http:LdapUserStoreConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/LdapUserStoreConfig) configurations.
-// Authorization is based on scopes. A scope maps to one or more groups.
-// Authorization can be enabled by setting the `string|string[]` type
-// configurations for `scopes` field.
+// Authorization is based on scopes. A scope maps to one or more groups. Authorization can be 
+// enabled by setting the `string|string[]` type configurations for `scopes` field.
```
Comment (praneesha) [inline examples/http-service-basic-auth-ldap-user-store/http_service_basic_auth_ldap_user_store.bal:16]:
```suggestion
// enabled by setting the `string|string[]` type configurations for the `scopes` field.
```

File: examples/http-service-jwt-auth/http_service_jwt_auth.bal [hunk 3]
```diff
@@ -9,12 +9,10 @@ listener http:Listener securedEP = new(9090,
...
-// The service can be secured with JWT Auth and can be authorized
-// optionally. JWT Auth can be enabled by setting the
-// [`http:JwtValidatorConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/JwtValidatorConfig) configurations.
-// Authorization is based on scopes. A scope maps to one or more groups.
-// Authorization can be enabled by setting the `string|string[]` type
-// configurations for `scopes` field.
+// The service can be secured with JWT Auth and can be authorized optionally. 
+// JWT Auth can be enabled by setting the [`http:JwtValidatorConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/JwtValidatorConfig) configurations.
+// Authorization is based on scopes. A scope maps to one or more groups. Authorization can be 
+// enabled by setting the `string|string[]` type configurations for `scopes` field.
```
Comment (praneesha) [inline examples/http-service-jwt-auth/http_service_jwt_auth.bal:15]:
```suggestion
// enabled by setting the `string|string[]` type configurations for the `scopes` field.
```

File: examples/http-service-oauth2/http_service_oauth2.bal [hunk 4]
```diff
@@ -9,12 +9,10 @@ listener http:Listener securedEP = new(9090,
...
-// The service can be secured with OAuth2 and by enforcing authorization
-// optionally. It can be enabled by setting the
-// [`http:OAuth2IntrospectionConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/OAuth2IntrospectionConfig) configurations.
-// Authorization is based on scopes. A scope maps to one or more groups.
-// Authorization can be enabled by setting the `string|string[]` type
-// configurations for `scopes` field.
+// The service can be secured with OAuth2 and by enforcing authorization optionally. 
+// It can be enabled by setting the [`http:OAuth2IntrospectionConfig`](https://docs.central.ballerina.io/ballerina/http/latest/records/OAuth2IntrospectionConfig) configurations.
+// Authorization is based on scopes. A scope maps to one or more groups. Authorization can be 
+// enabled by setting the `string|string[]` type configurations for `scopes` field.
```
Comment (praneesha) [inline examples/http-service-oauth2/http_service_oauth2.bal:15]:
```suggestion
// enabled by setting the `string|string[]` type configurations for the `scopes` field.
```

File: examples/http-client-bearer-token-auth/http_client_bearer_token_auth.md [hunk 5]
```diff
@@ -1,11 +1,12 @@
 # Client - Bearer Token Auth
 
-A client, which is secured with Bearer token auth can be used to connect to
-a secured service.<br/>
-The client is enriched with the `Authorization: Bearer <token>` header by
-passing the `http:BearerTokenConfig` for the `auth` configuration of the
+A client, which is secured with Bearer token auth can be used to connect to a secured service.
+
+The client is enriched with the `Authorization: Bearer <token>` header by passing the `http:BearerTokenConfig` for the `auth` configuration of the
 client.
```
Comment (ldclakmal) [inline examples/http-client-bearer-token-auth/http_client_bearer_token_auth.md:6]:
```suggestion
The client is enriched with the `Authorization: Bearer <token>` header by passing the `http:BearerTokenConfig` for the `auth` configuration of the client.
```

File: examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal [hunk 6]
```diff
@@ -1,16 +1,15 @@
 import ballerina/http;
 
 // HTTP version is set to 2.0.
-http:Client http2serviceClientEP =
-        check new ("http://localhost:7090", {httpVersion: "2.0"});
+http:Client http2serviceClientEP = check new ("http://localhost:7090", {httpVersion: "2.0"});
```
Comment (chamil321) [inline examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal:4]:
```suggestion
http:Client http2serviceClientEP = check new ("http://localhost:7090");
```

Comment (chamil321) [reply examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal:4]: Not needed as defaults to 2.0

File: examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal [hunk 7]
```diff
@@ -19,10 +18,9 @@ service / on new http:Listener(9090) {
 }
 
 // HTTP version is set to 2.0.
-listener http:Listener http2serviceEP = new (7090,
-    config = {httpVersion: "2.0"});
+listener http:Listener http2serviceEP = new (7090);
```
Comment (chamil321) [inline examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal:21]: What is the usage of `http2serviceEP`?

File: examples/http-caching-client/http_caching_client.bal [hunk 8]
```diff
@@ -10,15 +10,14 @@ import ballerina/http;
...
 // `CACHE_CONTROL_AND_VALIDATORS` (the default policy) and `RFC_7234`.
 
-http:Client cachingEP = checkpanic new ("http://localhost:8080",
-                        {cache: {enabled: true, isShared: true}});
+http:Client cachingEP = checkpanic new ("http://localhost:8080", 
+    cache = {enabled: true, isShared: true});
 service / on new http:Listener(9090) {
 
     resource function get cache(http:Request req)
             returns http:Response|error? {
```
Comment (chamil321) [inline examples/http-caching-client/http_caching_client.bal:18]: Move this to above line?

File: examples/http-caching-client/http_caching_client.client.out [hunk 9]
```diff
@@ -1,5 +1,5 @@
 # Invoke the service.
```
Comment (chamil321) [inline examples/http-caching-client/http_caching_client.client.out:1]: This line is redundant as we have added `Invoke the service as follows.` in the MD file before placing client.out files
Shall we fix them too?

Comment (ayeshLK) [reply examples/http-caching-client/http_caching_client.client.out:1]: Yeah sure.

File: examples/http-caching-client/http_caching_client.bal (hunk 8 above)
Comment (ayeshLK) [reply examples/http-caching-client/http_caching_client.bal:18]: I think it violates the character limit though. Will check again.

File: examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal (hunk 7 above)
Comment (ayeshLK) [reply examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal]: Fixed with [18d6eb246533e444b42727ad2010b376160c5cb1](https://github.com/ballerina-platform/ballerina-distribution/pull/3024/commits/18d6eb246533e444b42727ad2010b376160c5cb1)

File: examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal (hunk 6 above)
Comment (ayeshLK) [reply examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal]: Fixed with [18d6eb246533e444b42727ad2010b376160c5cb1](https://github.com/ballerina-platform/ballerina-distribution/pull/3024/commits/18d6eb246533e444b42727ad2010b376160c5cb1)

File: examples/http-caching-client/http_caching_client.client.out (hunk 9 above)
Comment (ayeshLK) [reply examples/http-caching-client/http_caching_client.client.out]: Fixed with [cddb02c200585b2c72d855a9b47b485ad2e6d0b5](https://github.com/ballerina-platform/ballerina-distribution/pull/3024/commits/cddb02c200585b2c72d855a9b47b485ad2e6d0b5)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**